Sources/GRPCCore/Call/Client/ClientRPCExecutionConfigurationCollection.swift

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+public struct ClientRPCExecutionConfigurationCollection: Sendable {
```

Comment:
I really don't like this name (I'm also not a huge fan of `ClientRPCExecutionConfiguration`). 

What we're really shadowing with these types is ServiceConfig and MethodConfig:
- https://github.com/grpc/grpc/blob/master/doc/service_config.md
- https://github.com/grpc/grpc-proto/blob/master/grpc/service_config/service_config.proto

At the moment `ClientRPCExecutionConfiguration` is a subset of MethodConfig which we'll likely need to expand and this type is a container for that.

I'm wondering if when we add a `Client` this should be nested within that type as `MethodConfigurationRegistry` or something and turn `ClientRPCExecutionConfiguration` into `Client.MethodConfiguration`.

WDYT?

Reply:
I agree, the name is bad. I'm unsure about renaming `ClientRPCExecutionConfiguration` to `MethodConfiguration` though, because, with the change you suggested in another comment to fall back to a service configuration by setting the descriptor's method name to `""`, the `MethodConfiguration` could also be referring to a service configuration, which sounds wrong. 

I think nesting the types inside the `Client` is fine though. But maybe it would make sense to have the same hierarchy `ServiceConfiguration`s containing `MethodConfiguration`s, like in the service_config proto file you shared?

Reply:
> I agree, the name is bad. I'm unsure about renaming `ClientRPCExecutionConfiguration` to `MethodConfiguration` though, because, with the change you suggested in another comment to fall back to a service configuration by setting the descriptor's method name to `""`, the `MethodConfiguration` could also be referring to a service configuration, which sounds wrong.

The configuration returned for a method name of `""` isn't service configuration: it would be the default method configuration for the service.

The service configuration is different still: it includes things like the retry throttle configuration _and_ any method configuration.

> I think nesting the types inside the `Client` is fine though. But maybe it would make sense to have the same hierarchy `ServiceConfiguration`s containing `MethodConfiguration`s, like in the service_config proto file you shared?

Yes, I think that'd be sensible.

Reply:
> it would be the default method configuration for the service
Fair enough :D

Should I leave the names as they are in this PR and make the whole change when I nest the types under `Client`, then?

Reply:
Yeah, we can leave them for now.